### PR TITLE
feat: add publish lockfile with supporting funcs

### PR DESCRIPTION
**Pull request by jmetz**

### CODE REVIEW

File: bioimageio_collection_backoffice/s3_client.py [hunk 1]
```diff
@@ -223,6 +223,9 @@ def _cp_dir(self, src: str, tgt: str):
 
         return objects
 
+    def rm(self, object: str):
+        self._client.remove_object(self.bucket, object)
```
Comment (FynnBe) [inline bioimageio_collection_backoffice/s3_client.py:227]: We should add the prefix here

**Review by FynnBe — CHANGES_REQUESTED** [review]

**Review by FynnBe — CHANGES_REQUESTED** [review]
hmm.. this lock does not prevent parallel publishing of different staged versions, right??

File: bioimageio_collection_backoffice/s3_client.py (hunk 1 above)
Comment (jmetz) [reply bioimageio_collection_backoffice/s3_client.py]: all good with the way I added the prefix @FynnBe?

File: bioimageio_collection_backoffice/remote_resource.py [hunk 2]
```diff
@@ -539,6 +539,18 @@ def _set_status(self, value: StagedVersionStatus):
...
+        """Creates publish lock in DB"""
+        self.client.put_and_cache(self.publish_lockfile_path, b" ")
+
+    def unlock_publish(self):
+        """Releases publish lock in DB"""
+        self.client.rm(self.publish_lockfile_path)
+
+    @property
+    def publish_lockfile_path(self):
+        return f"{self.folder}lock-publish"
```
Comment (FynnBe) [inline bioimageio_collection_backoffice/remote_resource.py:552]: this should now lock for the resource "concept" (all versions)
```suggestion
        return f"{self.concept.folder}lock-publish"
```

**Pull request merged**